addons/delivery/views/report_shipping.xml

```diff
@@ -21,5 +21,15 @@
                 </td>
             </xpath>
         </template>
+        <template id="report_shipping3" inherit_id="stock.report_delivery_document">
+            <xpath expr="//th[@name='td_sched_date_h']" position="after">
+                <th><strong>Weight</strong></th>
+            </xpath>
+            <xpath expr="//td[@name='schedule_date']" position="after">
+                <td>
+                    <span t-field="o.weight"/>
+                </td>
+            </xpath>
+        </template>
```

Comment:
cannot we check this condition  in stock/views/report_delivery.xml
<t t-if="'weight' in o._columns.keys()">
      <td>
                    <span t-field="o.weight"/>
        </td>
</t> instead of inheriting it in delivery module ?